circuits/src/mpc_circuits/settle.rs

```diff
@@ -0,0 +1,214 @@
+            match_res.quote_amount.clone(),
+        ],
+    );
+
+    let party0_buy = amounts.remove(0);
+    let party0_sell = amounts.remove(0);
+
+    // Update the balances of the two parties
+    let party0_buy_balance =
+        &mut party0_new_shares.balances[party0_settle_indices.balance_receive as usize];
+    party0_buy_balance.amount = &party0_buy_balance.amount + &party0_buy;
+
+    let party1_buy_balance =
+        &mut party1_new_shares.balances[party1_settle_indices.balance_receive as usize];
+    party1_buy_balance.amount = &party1_buy_balance.amount + &party0_sell;
+
+    let party0_sell_balance =
+        &mut party0_new_shares.balances[party0_settle_indices.balance_send as usize];
+    party0_sell_balance.amount = &party0_sell_balance.amount - &party0_sell;
+
+    let party1_sell_balance =
+        &mut party1_new_shares.balances[party1_settle_indices.balance_send as usize];
+    party1_sell_balance.amount = &party1_sell_balance.amount - &party0_buy;
+
+    (party0_new_shares, party1_new_shares)
+}
+
+#[cfg(test)]
+mod test {
+    use std::iter;
+
+    use ark_mpc::{PARTY0, PARTY1};
+    use circuit_types::{
+        order::OrderSide,
+        r#match::MatchResult,
+        traits::{BaseType, MpcBaseType, MpcType},
+        SizedWalletShare,
+    };
+    use constants::{Scalar, MAX_BALANCES, MAX_ORDERS};
+    use rand::{thread_rng, Rng};
+    use renegade_crypto::fields::scalar_to_biguint;
+    use test_helpers::mpc_network::execute_mock_mpc;
+
+    use crate::{
+        mpc_circuits::settle::settle_match,
+        zk_circuits::{
+            valid_commitments::OrderSettlementIndices,
+            valid_match_settle::test_helpers::apply_match_to_shares,
+        },
+    };
+
+    /// The parameterization of a test
+    #[derive(Clone)]
+    struct SettlementTest {
+        /// The match result to settle
+        match_res: MatchResult,
+        /// The shares of the first party before settlement
+        party0_pre_shares: SizedWalletShare,
+        /// The indices of the first party's order and balances to settle
+        party0_indices: OrderSettlementIndices,
+        /// The shares of the second party before settlement
+        party1_pre_shares: SizedWalletShare,
+        /// The indices of the second party's order and balances to settle
+        party1_indices: OrderSettlementIndices,
+        /// The shares of the first party after settlement
+        party0_post_shares: SizedWalletShare,
+        /// The shares of the second party after settlement
+        party1_post_shares: SizedWalletShare,
+    }
+
+    /// Get a dummy set of inputs for a settlement circuit
+    fn generate_test_params() -> SettlementTest {
+        let mut rng = thread_rng();
+        let quote_mint = scalar_to_biguint(&Scalar::random(&mut rng));
+        let base_mint = scalar_to_biguint(&Scalar::random(&mut rng));
+
+        let match_res = MatchResult {
+            quote_mint: quote_mint.clone(),
+            base_mint: base_mint.clone(),
```

Comment:
are these two also unused? if not, then how does using `random_shares` and `random_indices` work for the test (i would think the indices would need to point to shares that line up with the correct quote/base mints)

Reply:
Nothing about the act of settling guarantees that the mints are correct. This is done elsewhere: concretely in the `VALID COMMITMENTS` circuit. So such indices are guaranteed to be correct for a commitment linked order before they are used. The way the tests use them is misleading, you're right, I'll mark that